test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/persistence/WorkbenchPersistenceIntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+                workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername(), workbenchWithKieServerScenario.getWorkbenchDeployment().getPassword());
+    }
+
+    @After
+    public void tearDown() {
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getKieServerDeployment());
+        workbenchWithKieServerScenario.undeploy();
+        gitProvider.deleteGitRepository(workbenchWithKieServerScenario.getNamespace());
+    }
+
+    @Test
+    public void testWorkbenchControllerPersistence() {
+        workbenchClientProvider.createOrganizationalUnit(ORGANIZATION_UNIT_NAME, workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername());
+        workbenchClientProvider.cloneRepository(ORGANIZATION_UNIT_NAME, REPOSITORY_NAME, gitProvider.getRepositoryUrl(workbenchWithKieServerScenario.getNamespace()));
+        workbenchClientProvider.deployProject(REPOSITORY_NAME, PROJECT_NAME);
+
+        KieServerInfo serverInfo = kieServerClientProvider.getKieServerClient().getServerInfo().getResult();
+        String kieServerLocation = serverInfo.getLocation();
+
+        kieControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, PROJECT_NAME, PROJECT_VERSION, KieContainerStatus.STARTED);
+
+        verifyOneServerTemplateWithContainer(kieServerLocation, CONTAINER_ID);
+
+        workbenchWithKieServerScenario.getWorkbenchDeployment().scale(0);
+        workbenchWithKieServerScenario.getWorkbenchDeployment().waitForScale();
+        workbenchWithKieServerScenario.getWorkbenchDeployment().scale(1);
+        workbenchWithKieServerScenario.getWorkbenchDeployment().waitForScale();
+
+        verifyOneServerTemplateWithContainer(kieServerLocation, CONTAINER_ID);
+    }
+
+    @Test
+    public void testWorkbenchProjectPersistence() {
+        workbenchClientProvider.createOrganizationalUnit(ORGANIZATION_UNIT_NAME, workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername());
+        workbenchClientProvider.createRepository(ORGANIZATION_UNIT_NAME, REPOSITORY_NAME);
+        workbenchClientProvider.createProject(REPOSITORY_NAME, PROJECT_GROUP_ID, PROJECT_NAME, PROJECT_VERSION);
+
+        assertRepositoryAndProjectExists(REPOSITORY_NAME, PROJECT_NAME);
+
+        workbenchWithKieServerScenario.getWorkbenchDeployment().scale(0);
+        workbenchWithKieServerScenario.getWorkbenchDeployment().waitForScale();
+        workbenchWithKieServerScenario.getWorkbenchDeployment().scale(1);
+        workbenchWithKieServerScenario.getWorkbenchDeployment().waitForScale();
+
+        assertRepositoryAndProjectExists(REPOSITORY_NAME, PROJECT_NAME);
+
+        workbenchClientProvider.deployProject(REPOSITORY_NAME, PROJECT_NAME);
+
+        workbenchWithKieServerScenario.getWorkbenchDeployment().scale(0);
```

Comment:
This scaleDownAndUp code is repeated three times. I'd extract that to helper method.